apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html

```diff
@@ -1,61 +1,117 @@
+    <div class="mb-4 d-flex gap-4 justify-content-between">
+      <div class="input-group">
+        <label class="input-group-text" for="search-styling">
+          <i class="mx-1 icon-search" aria-label="Search"></i>
+        </label>
+        <input class="form-control" formControlName="search" type="text" id="search-styling" placeholder="Search for styling variable" />
+      </div>
+      <div class="form-check form-switch">
+        <input class="form-check-input" type="checkbox" id="themeOnly" formControlName="themeOnly">
+        <label class="form-check-label text-nowrap" for="themeOnly">
+          Show only theming variable
+        </label>
+      </div>
+    </div>
+    <div class="btn-group" role="group">
+      @for (group of ['category', 'type', 'component']; track group) {
+        <input type="radio" class="btn-check" [id]="group" [value]="group" autocomplete="off" formControlName="groupBy">
+        <label class="btn btn-outline-primary" [for]="group">Grouped by {{ group }}</label>
+      }
     </div>
   </form>
   <form [formGroup]="form.controls.variables">
-    <ul class="d-grid w-100 overflow-hidden p-0 border-top border-start border-light" [style.grid-template-columns]="'repeat(auto-fill, minmax(300px, 1fr))'">
-      @for (themeVar of filteredThemeVariables(); track themeVar.name) {
-        <li class="form-group d-flex gap-4 border-bottom border-end border-light p-7">
-          <div class="d-flex flex-column gap-2 w-100">
-            <label class="form-label text-nowrap fw-semibold" [for]="themeVar.name">{{ themeVar | variableLabel }}</label>
-            <div class="d-flex gap-4 align-items-center position-relative">
-              @if (themeVar.type === 'color') {
-                <input [style.width]="'40px'" class="form-control form-control-color" type="color" [value]="resolvedVariables()[themeVar.name] | color" (change)="onColorChange(themeVar.name, $event)" />
+    <div ngbAccordion [closeOthers]="true" #acc="ngbAccordion">
+      @for (group of groupedVariables(); track group.name) {
+        <div ngbAccordionItem>
+          <h3 ngbAccordionHeader>
+            <button ngbAccordionButton>
+              @if (group.isPalette && group.defaultVariable) {
+                <div (click)="$event.stopPropagation();">
+                  <input [style.width]="'40px'" class="form-control form-control-color" [id]="group.name" type="color" [value]="resolvedVariables()[group.defaultVariable.name] | hexColor" (change)="onPaletteChange(group, $event)" />
+                </div>
+                <label class="form-label text-nowrap fw-semibold my-0 ms-4" [for]="group.name">{{ group.name }}</label>
+                @if (form.value.variables?.[group.defaultVariable.name] !== group.defaultVariable.defaultValue) {
+                  <i class="form-text text-warning icon-undo position-absolute end-0 me-10" ngbTooltip="Reset all colors of {{group.name}}" (click)="onPaletteReset(group, $event)"></i>
```

Comment:
I don't think it's a good practice in terms of accessibility to have a clickable element inside the header of the accordion which is already clickable